src/main/java/org/jenkinsci/plugins/casc/ObsoleteConfigurationMonitor.java

```diff
@@ -51,7 +51,8 @@ public Error(Source source, String message) {
     }
 
     public String getCss() {
```

Comment:
still not restricted

Reply:
Do we have any way to do the inverse logic ? i.e Restrict everything and only opt-in for methods which are part of the "public API"

Reply:
Unfortunately no. We were talking about it with @jglick a while ago. 
Will create a JIRA ticket as a follow-up and handle APIs here in the bulk mode

Reply:
> Netbeans-style API plugins

Is there any doc on Netbeans plugin model ? 
I really would like to see this improved, Jenkins is missing a strong plugin model, and I don't think it would be trivial to use OSGi here :-\